waspc/e2e-test/Util.hs

```diff
@@ -0,0 +1,82 @@
+module Util
+  ( newCommand,
+    newAndCompileCommand,
+    newAndCompileAndMigrateCommand,
+    runGoldenTest,
+  )
+where
+
+import Control.Monad (filterM)
+import Data.List (intercalate, sort)
+import Data.Text (pack, replace, unpack)
+import System.Directory (doesFileExist)
+import System.Directory.Recursive (getDirFiltered)
+import System.FilePath (joinPath, makeRelative, takeFileName, (</>))
+import System.Process (callCommand)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Golden (goldenVsFileDiff)
+
+-- TODO: Switch over from FilePath to StrongPath eventually
+
+-- TODO: Handle shell command descriptions/invocations and file manipulation in some better way (and more platform agnostic?)
+-- NOTE: Using `wasp-cli` herein so we can assume using latest `stack install` in CI and locally. Better way?
+newCommand :: String -> String
+newCommand projectName =
+  intercalate " && " ["wasp-cli new " ++ projectName, "cp " ++ ("overlays" </> ".gitignore") ++ " " ++ projectName]
+
+newAndCompileCommand :: String -> String
+newAndCompileCommand projectName =
+  intercalate " && " [newCommand projectName, "cd " ++ projectName, "wasp-cli compile"]
+
+newAndCompileAndMigrateCommand :: FilePath -> FilePath -> String -> String
+newAndCompileAndMigrateCommand _cwd overlayFp projectName =
```

Comment:
Yeah I kinda made that up, since to me they are not fixtures in the traditional sense how I'm (abusing? lol) using them... I am using them sometimes to overwrite an entire file (like overlays/.gitignore), sometimes to append content to another file (like overlays/entity.txt), so the idea of "overlaying files/contents onto existing ones" was kinda where I was going. This whole area can probably be cleaned up some.

Reply:
Can you maybe help me understand better what these are used for? What are the use cases you use them for, in more details (ELI5).

Reply:
The thinking here was having an easy way to add discrete things to Wasp files, like entities. This is in contrast to having a full Wasp file we overwrite. I thought the applying patches approach was a bit more flexible and more maintainable.

Reply:
Ah got it! Ok, so we can ensure that for example Wasp file has an entity? Makes sense! Might be interesting to find a better name, if we can.

Reply:
Yeah agreed,  not sure what the convention here is, will think on it and at least better comment the purpose in meantime 🤔 👍🏻